_posts/2022-02-11-helm-multiple-worker-groups.md

```diff
@@ -0,0 +1,124 @@
+-  Next, we’re going to update the section in the file called ‘additional_worker_groups’. The section looks like this:
+```yaml
+additional_worker_groups: # Additional groups of workers to create
+# - name: high-mem-workers  # Dask worker group name.
+#   resources:
+#     limits:
+#       memory: 32G
+#     requests:
+#       memory: 32G
+# ... 
+# (Defaults will be taken from the primary worker configuration)
+```
+- Now we’re going to edit the section to look like this:
+```yaml
+additional_worker_groups: # Additional groups of workers to create
+- name: gpu-workers  # Dask worker group name.
+  replicas: 1
+  image:
+    repository: rapidsai/rapidsai-core
+    tag: 21.12-cuda11.5-runtime-ubuntu20.04-py3.8
+    dask_worker: dask-cuda-worker
+  extraArgs:
+    - --resources
+    - "GPU=1"
+  resources:
+    limits:
+      nvidia.com/gpu: 1
+```
+- Now we can update our deployment with our new values.yaml
+```console
+helm upgrade -f my-values.yaml my-dask dask/dask
+```
+- Again, if you have kubectl installed, you can run ‘kubectl get all -n default’, and you’ll see our new GPU worker pod running: `kubectl get all -n default`
+<img src="/images/gpu-worker-dask-cluster.png" alt="Dask Cluster Installed with Helm with a GPU worker" width="653" height="428">
+- Now we can open up a jupyter notebook or any editor to write some code.
+
+## Scaling the workers Up/Down
+```python
+from dask_kubernetes import HelmCluster
+cluster = HelmCluster(release_name="my-dask")
+cluster
+```
+<img src="/images/dask-cluster-four-workers.png" alt="Dask Cluster with four workers" width="1002" height="659">
+
+```python
+cluster.scale(5)  # scale the default worker group from 3 to 5 workers 
+cluster
+```
+<img src="/images/dask-cluster-six-workers.png" alt="Dask Cluster with six cluster" width="1002" height="802">
+
+```python
+cluster.scale(2, worker_group = ”gpu-workers”)  # scale the GPU worker group from 1 to 2 workers
+cluster
+```
+<img src="/images/dask-cluster-seven-workers.png" alt="Dask Cluster with seven cluster" width="992" height="845">
+
+## Example: Finding the average New York City taxi trip distance in April 2020
+```python
+import dask.dataframe as dd
+import dask
+
+link = "https://s3.amazonaws.com/nyc-tlc/trip+data/yellow_tripdata_2020-04.csv"
+ddf = dd.read_csv(link, assume_missing=True)
+avg_trip_distance = ddf['trip_distance'].mean().compute()
+print(f"In January 2021, the average trip distance for yellow taxis was {avg_trip_distance} miles.")
+
+with dask.annotate(resources={'GPU': 1}):
+    import dask_cudf, cudf
+    dask_cdf = ddf.map_partitions(cudf.from_pandas)
+    avg_trip_distance = dask_cdf['trip_distance'].mean().compute()
+    print(f"In January 2021, the average trip distance for yellow taxis was {avg_trip_distance} miles.")
+```
+
+## Closing
+That’s it! We’ve deployed Dask with Helm, created an additional GPU worker type, and used our workers to run an example calculation using the NY Taxi dataset.
```

Comment:
Maybe wrap up with a little more of a why.

What is new? Why did we add additional worker groups to the chart? Why did we update the `HelmCluster` to scale them. 

It would also be nice to hint at worker groups being a thing we want to think about more in the future. Maybe mention that until now most Dask deployments have homogenous workers, but as users are pushing Dask further there is a growing demand for heterogenous clusters with special purpose workers. We want to add worker groups throughout Dask.